Joinmarket-messaging-protocol.md

```diff
@@ -200,4 +201,66 @@ Commitment data for default PoDLE construct:
 
+
+Fidelity Bonds
+==========
+
+JoinMarket version 0.9 implements fidelity bonds, for a full writeup of whys and hows of the feature see [the design document](https://gist.github.com/chris-belcher/18ea0e6acdb885a2bfbdee43dcd6b5af/).
+
+Makers who have a fidelity bond will send a proof of it to takers in response to an !orderbook message along with the maker's offers.
+
+The fidelity bond proof message contains everything needed to prove to the taker that the fidelity bond UTXO is real, that it has a certain value, and that it is genuinely possessed by the maker.
+To keep the size down the fidelity bond proof message is a base64-encoded binary data blob.
+
+The binary blob is made up of concatenating various parameters (size in bytes on the line below):
+
+```
+nick_sig + cert_sig + cert_pubkey + cert_expiry + utxo_pubkey + txid + vout + timelock
+72       + 72       + 33          + 2           + 33          + 32   + 4    + 4 = 252 bytes
+```
+
+The command name "tbond" means "time locked bond", one day there might be a burned coins fidelity bond too, but probably not because timelocked bonds 
+
+### UTXO data
+To prove that the UTXO exists, the proof contains the `txid` and `vout` of the UTXO, these can be used to lookup in a full node's UTXO set to check that the UTXO exists, is confirmed in the strongest chain and is unspent.
+The proof also contains a `utxo_pubkey` and `locktime` which together are enough to construct a redeemscript for a timelocked address.
+The UTXO set query will give the verifier information about the UTXO's scriptPubKey and allow him to verify that the UTXO is indeed a timelocked address.
+The UTXO set query will also tell the verifier how many bitcoins are locked in the address, as well as the confirmation time of the transaction.
+
+The receiver must check that the given (`txid`, `vout`) has not already been seen. A UTXO must only be used at most once for a fidelity bond in a given orderbook.
+If a UTXO does arrive at the taker a second time, the taker should just pick one (it doesnt matter which one, either way removes the incentive to announce UTXOs more than once).
+
+### Two signatures
+
+There are two signatures in the proof.
+
+Diagram of the two signatures:
+
+`Fidelity bond keypair ----signs----> certificate ----signs----> IRC nicknames`
+
+#### Certificate signature
+In order to allow holding the fidelity bond UTXOs in cold storage, there is an intermediate keypair called the certificate.
+A fidelity bond privkey can sign a certificate message and transfer the `utxo_pubkey`, `cert_pubkey`, `cert_sig` and `cert_expiry` to the online computer.
+
+The certificate message is defined as `'fidelity-bond-cert|' + cert_pubkey + '|' + cert_expiry` encoded as bytes where + denotes concatenation.
+
+The certificate expiry `cert_expiry` is the number of the 2016-block period after which the certificate is no longer valid.
```

Comment:
Maybe write explicitly how to calculate this number (I guess it's *fairly* obvious but if you were to reproduce you might get a fencepost error).